### PR TITLE
Add ProviderAws Tests

### PR DESCRIPTION
**Pull request by erikerikson**
Add tests for credential loading by ProviderAws.

Add those tests to the "all tests" script.

### CODE REVIEW

File: tests/tests/classes/ProviderAws.js [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    os          = require('os'),
+    testUtils   = require('../../test_utils'),
+    testConfig  = require('../../config'),
+    Serverless  = require('../../../lib/Serverless.js'),
+    ProviderAws = require('../../../lib/ProviderAws.js'),
+    regions = [
+        'us-east-1',      // California
+        'us-west-2',      // Oregon
+        'eu-west-1',      // Ireland
+        'ap-northeast-1'  // Tokyo
```
Comment (dherault) [inline tests/tests/classes/ProviderAws.js]: 'eu-central-1', // Frankfurt

Comment (erikerikson) [inline tests/tests/classes/ProviderAws.js]: thanks!

**Comment by eahefnawy** [comment]
@erikerikson you've probably refactored this PR many times cause of our slowness (really sorry :disappointed: ) ... So I wanna merge this right away. Is it good to go now? :)

**Comment by erikerikson** [comment]
@eahefnawy @ac360 done.  No problem.  This isn't the highest priority by a long shot.  Plus, easy merges.  :)

**Comment by eahefnawy** [comment]
Awesome! Thanks a lot! :blush:

**Pull request merged**